chore: Make it easier to override the service host in generated libraries

dazuma opened the pull request:
Synth hack that adds `service_address` and `service_port` parameters to generated client constructors. This makes it easier/possible to point at a different backend. (Previously, you had to subclass the client class and change the constant, and it wasn't even possible to customize the corresponding LRO endpoint.)

This WIP shows the result for vision as an example. If people are good with how it looks, I'll add the remaining libraries.

blowmage commented on google-cloud-vision/lib/google/cloud/vision/v1.rb:
```diff
@@ -133,10 +133,16 @@ module ImageAnnotator
           #   An optional proc that intercepts any exceptions raised during an API call to inject
           #   custom error handling.
           def self.new \
+              service_address: nil,
```
Is it possible to list these arguments in the same order that they are listed in the documentation?

dazuma replied:
Done.

dazuma commented:
This is now final. I included a few more fixes in the synth script and replicated it across all libraries.

Final changes:
* All `google-gax` dependencies are now `~> 1.7`. For generated libraries that synth the gemspec, the synth script has been updated to fix that dependency. For handwritten libraries, the gemspec has been edited manually.
* Added `service_address` and `service_port` parameters to the constructors for all *generated* client classes (including the generated convenience module methods). This includes client classes in the generated low-level layers of handwritten libraries. I did *not*, however, include handwritten edits to expose these parameters in the handwritten layers of handwritten libraries. That will come separately.
* Removed redundant exception-related synth hacks (since the fix was pushed upstream into artman) so that synthtool no longer warns about it.

This PR does *not* include the results of rerunning synthtool. I'm going to let autosynth do that and open its normal PRs. That way, it should be easier for us to review the results library-by-library.

dazuma commented:
Set the type to `chore` because this doesn't actually implement the feature itself, but just the code generator that generates the feature.
